astropy/stats/funcs.py

```diff
@@ -790,3 +792,322 @@ def mad_std(data):
+        s = 0.
+        # NOTE: the upper limit of this sum is taken from Stephens 1965
+        for t in xrange(int(np.floor(N*(1-D)))+1):
+            term = T(t)*comb(N,t)*(1-D-t/float(N))**(N-t-1)
+            s += term
+        return s
+    else:
+        z = D*np.sqrt(N) 
+        S1 = 0.
+        term_eps = 1e-12
+        abs_eps = 1e-100
+        for m in itertools.count(1):
+            T1 = 2.*(4.*m**2*z**2-1.)*np.exp(-2.*m**2*z**2)
+            so = S1
+            S1 += T1
+            if np.abs(S1-so)/(np.abs(S1)+np.abs(so))<term_eps or np.abs(S1-so)<abs_eps:
+                break
+        S2 = 0.
+        for m in itertools.count(1):
+            T2 = m**2*(4.*m**2*z**2-3.)*np.exp(-2*m**2*z**2)
+            so = S2
+            S2 += T2
+            if np.abs(S2-so)/(np.abs(S2)+np.abs(so))<term_eps or np.abs(S1-so)<abs_eps:
+                break
+        return S1 - 8*D/(3.*np.sqrt(N))*S2
+
+def kuiper(data, cdf=lambda x: x, args=()):
+    """Compute the Kuiper statistic.
+    
+    Use the Kuiper statistic version of the Kolmogorov-Smirnov test to 
+    find the probability that something like data was drawn from the 
+    distribution whose CDF is given as cdf.
+    
+    Parameters
+    ----------
+    data : array-like
+        The data values.
+    cdf : callable
+        A callable to evaluate the CDF of the distribution being tested
+        against. Will be called with a vector of all values at once.
+    args : list-like, optional
+        Additional arguments to be supplied to cdf.
+
+    Returns
+    -------
+    D : float
+        The raw statistic.
+    fpp : float
+        The probability of a D this large arising with a sample drawn from
+        the distribution whose CDF is cdf.
+
+    Notes
+    -----
+    The Kuiper statistic resembles the Kolmogorov-Smirnov test in that 
+    it is nonparametric and invariant under reparameterizations of the data. 
+    The Kuiper statistic, in addition, is equally sensitive throughout 
+    the domain, and it is also invariant under cyclic permutations (making 
+    it particularly appropriate for analyzing circular data). 
+
+    Returns (D, fpp), where D is the Kuiper D number and fpp is the 
+    probability that a value as large as D would occur if data was 
+    drawn from cdf.
+
+    Warning: The fpp is calculated only approximately, and it can be 
+    as much as 1.5 times the true value.
+
+    Stephens 1970 claims this is more effective than the KS at detecting 
+    changes in the variance of a distribution; the KS is (he claims) more 
+    sensitive at detecting changes in the mean.
+
+    If cdf was obtained from data by fitting, then fpp is not correct and 
+    it will be necessary to do Monte Carlo simulations to interpret D. 
+    D should normally be independent of the shape of CDF.
+
+    """
+
+    # FIXME: doesn't work for distributions that are actually discrete (for example Poisson).
```

Comment:
Do you intend to fix this before the PR is merged? Otherwise I would suggest to put that as a warning in the docstring, too.